sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -198,6 +200,9 @@ static void _process_coap_pdu(sock_udp_t *sock, sock_udp_ep_t *remote,
         /* normal request */
         else if (coap_get_type(&pdu) == COAP_TYPE_NON
                 || coap_get_type(&pdu) == COAP_TYPE_CON) {
+            #if IS_USED(MODULE_NANOCOAP_PKT_IPV6_ADDRESS)
+                memcpy(&pdu.source_address, &aux->local.addr.ipv6, sizeof(ipv6_addr_t));
+            #endif
```

Comment:
I wasn't entirely sure if this is actually working but I found something similar in the [existing codebase](https://github.com/RIOT-OS/RIOT/blob/08d86295d230e586a08aceb6e5bdf4438ff3b5d6/tests/gnrc_sock_ip/main.c#L115). In any case, compiling the `gcoap` example with the added submodule works but actual testing (perhaps in the context of our `wot` implementation) is still needed.